kv/kvserver: TestLeasePreferencesDuringOutage failed

kv/kvserver.TestLeasePreferencesDuringOutage [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10573809?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10573809?buildTab=artifacts#/) on release-22.2 @ [8055184fc991424d1d0eefd7ccf703948b8de3ee](https://github.com/cockroachdb/cockroach/commits/8055184fc991424d1d0eefd7ccf703948b8de3ee):


```
=== RUN   TestLeasePreferencesDuringOutage
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestLeasePreferencesDuringOutage4182034325
    test_log_scope.go:79: use -show-logs to present logs inline
    client_lease_test.go:1028: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/client_lease_test.go:1028
        	Error:      	Received unexpected error:
        	            	replica [n1,s1,r53/1:/{Table/100-Max}] does not have the range lease: [NotLeaseHolderError] lease acquisition attempt lost to another lease, which has expired in the meantime; r53: replica (n1,s1):1 not lease holder; lease holder unknown
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Enqueue
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:3644
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestLeasePreferencesDuringOutage
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/client_lease_test.go:1026
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) replica [n1,s1,r53/1:/{Table/100-Max}] does not have the range lease
        	            	Wraps: (3) [NotLeaseHolderError] lease acquisition attempt lost to another lease, which has expired in the meantime; r53: replica (n1,s1):1 not lease holder; lease holder unknown
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.NotLeaseHolderError
        	Test:       	TestLeasePreferencesDuringOutage
        	Messages:   	failed to enqueue replica for replication
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestLeasePreferencesDuringOutage4182034325
--- FAIL: TestLeasePreferencesDuringOutage (6.38s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeasePreferencesDuringOutage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-28862

--- a/pkg/kv/kvserver/client_lease_test.go
+++ b/pkg/kv/kvserver/client_lease_test.go
@@ -947,6 +947,13 @@ func TestLeasePreferencesDuringOutage(t *testing.T) {
 		locality("us", "mi"),
 		locality("us", "mi"),
 	}
+	// Disable expiration based lease transfers. It is possible that a (pseudo)
+	// dead node acquires the lease and we are forced to wait out the expiration
+	// timer, if this were not set.
+	settings := cluster.MakeTestingClusterSettings()
+	sv := &settings.SV
+	kvserver.TransferExpirationLeasesFirstEnabled.Override(ctx, sv, false)
+	kvserver.ExpirationLeasesOnly.Override(ctx, sv, false)
 	for i := 0; i < numNodes; i++ {
 		serverArgs[i] = base.TestServerArgs{
 			Locality: localities[i],

